### ISSUE TITLE
Facebook redirectUri required https

### ISSUE THREAD

**Issue by seanpconkie** [aspnet/Docs#9549]
Have integrated facebook external login but https is not enforced in my app in production, it is handled by load balancer which maintains certificate etc.  Is there a way for me to edit the redirect Uri sent to facebook to reference https instead of http?  Or am I doing it all wrong?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d1612f19-f267-493a-c230-2ca18e4eb50b
* Version Independent ID: d312db0f-6cf1-87ca-09b0-604a66e18a07
* Content: [Facebook, Google, and external provider authentication in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/authentication/social/?view=aspnetcore-2.1)
* Content Source: [aspnetcore/security/authentication/social/index.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/security/authentication/social/index.md)
* Product: **aspnet-core**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
Hello @seanpconkie ... Possibly the scheme isn't being preserved. Use the Forwarded Headers Middleware with the app. See: https://docs.microsoft.com/aspnet/core/host-and-deploy/proxy-load-balancer

If that doesn't do the trick, open an issue on the aspnet/Home issues to chat with the engineers. Leave this issue open. I want to get that topic cross-linked in here.

**Comment by guardrex** [comment]
It's a common scenario with proxies and LBs that the scheme ends up in the request header ... thus the app doesn't get it without the Forwarded Headers Middleware. If you look at the request that's reaching the app, and you see an `X-Forwarded-Proto` header with `https`, then that's likely what's happening. Run the little code snippet from the top of the troubleshooting section of that topic. The app can show you what it's seeing.

https://docs.microsoft.com/aspnet/core/host-and-deploy/proxy-load-balancer#troubleshoot

If your issue turns out to be something else, then open an issue over on the aspnet/Home repo for the engineers to take a look.

**Comment by seanpconkie** [comment]
Tried some of the suggestions in troubleshooting and have following output;
`Request Method: GET
Request Scheme: https
Request Path: /
Request Headers:
Connection: Keep-Alive
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Cookie: #########################
Host: ###########################.elb.amazonaws.com
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.77 Safari/537.36
X-Forwarded-For: ##############
X-Forwarded-Port: 443
X-Amzn-Trace-Id: Root=##################
upgrade-insecure-requests: 1
MS-ASPNETCORE-TOKEN: #################
X-Original-For: #############
X-Original-Proto: http

Request RemoteIp: 172.31.2.183`

There is no x-forwarded-proto, which I think is correct as x-original-proto is http - in this example LB redirected from http to https.  Request Scheme shows https.

However, I still get the http redirect uri for external login providers.  Is this another issue I should be raising in aspnet/Home repo?

**Comment by guardrex** [comment]
@Tratcher Would you like to have @seancpeters open this on aspnet/Home, or take a look here at his issue?

**Comment by Tratcher** [comment]
The forwarders look right. Can you share a Fiddler trace file showing the scenario?

**Comment by seancpeters** [comment]
I think @guardrex meant @seanpconkie in their previous post, not me (@seancpeters) :)

**Comment by guardrex** [comment]
Sorry about that @seancpeters ... GH/browser auto-pulled it based on similar first few letters and I didn't notice.

**Comment by ajitgoel** [comment]
I have the same issue and here is my troubleshooting output. What am I missing? 

Header: "X-Client-IP": ["76.187.198.247"]
Header: "X-Client-Port": ["51335"]
Header: "Upgrade-Insecure-Requests": ["1"]
Header: "DNT": ["1"]
Header: "X-WAWS-Unencoded-URL": ["/Identity/Account/Login"]
Header: "CLIENT-IP": ["76.187.198.247:51335"]
Header: "X-ARR-LOG-ID": ["3b69d760-03e7-4199-bec4-38ff77055413"]
Header: "DISGUISED-HOST": ["simplerproductsscrubber.azurewebsites.net"]
Header: "X-SITE-DEPLOYMENT-ID": ["SimplerProductsScrubber"]
Header: "WAS-DEFAULT-HOSTNAME": ["simplerproductsscrubber.azurewebsites.net"]
Header: "X-Original-URL": ["/Identity/Account/Login"]
Header: "X-Forwarded-For": ["76.187.198.247:51335"]
Header: "X-ARR-SSL": ["2048|256|C=US, O=DigiCert Inc, CN=DigiCert SHA2 Secure Server CA|C=US, S=Washington, L=Redmond, O=Microsoft Corporation, CN=*.azurewebsites.net"]
Header: "X-Forwarded-Proto": ["https"]
Header: "X-AppService-Proto": ["https"]
Request RemoteIp: "::ffff:172.16.1.1"
Request Method: "GET"
Request Scheme: "http"
Request Path: "/Identity/Account/Login"
Header: "Connection": ["close"]
Header: "Accept": ["text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3"]
Header: "Accept-Encoding": ["gzip, deflate, br"]
Header: "Accept-Language": ["en-US,en;q=0.9"]
Header: "Cookie": ["ARRAffinity=152c130e21c95ce31be52418aed58ed4a1114b560e108246b2120e2d4dbf27ee; .AspNetCore.Antiforgery.nixphHDAMN4=CfDJ8G1Jn3njIA5IoKC-W8RHjabWwnkwCrPq4ZnU7-ZRlTXbuf8kfpKPQACS5HEylcqol59j-9GJ4AzKFgirMIn8yclO5QSucBnlED9aKjQgAlRrkuZmIZeu8VKT9oOA1V_dvEpjhDoqKxxWrRpfVwST6hU"]
Header: "Host": ["simplerproductsscrubber.azurewebsites.net"]
Header: "Max-Forwards": ["10"]
Header: "Referer": ["https://simplerproductsscrubber.azurewebsites.net/Dashboard"]
Header: "User-Agent": ["Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/74.0.3729.169 Safari/537.36"]
Faceb

**Comment by Tratcher** [comment]
Yours looks the same as the others.
```
Header: "X-Forwarded-Proto": ["https"]
Request Scheme: "http"
```
Is this the App Service Linux host?
See https://github.com/aspnet/AspNetCore/issues/4135#issuecomment-493592243

**Comment by ajitgoel** [comment]
Thanks Chris, yes my app service has a Linux host.

**Comment by guardrex** [comment]
@Tratcher Do you want https://github.com/aspnet/AspNetCore/issues/4135 covered in the proxy/LB topic? If so, I'll open an issue. If so, confirm if the following is correct:

* Description of problem based on your opening remarks of the engineering issue. 
* `ASPNETCORE_FORWARDEDHEADERS_ENABLED` set to `true` in app settings. If this instruction can drop at 3.0 RTM with new container images, I'll open a 2nd follow-up issue to remove this remark on adding the app setting at 3.0 RTM.
* The `ConfigureServices` code of the engineering issue (the version that checks the env var).

**Comment by Tratcher** [comment]
@guardrex yes please. @bradygaster is writing up a blog post for 2.x folks and it has been fixed in 3.0.

Setting ASPNETCORE_FORWARDEDHEADERS_ENABLED is a point in time issue until they update the container images. @bradygaster that image update isn't 3.0 specific, right?

**Comment by bradygaster** [comment]
That is correct @Tratcher - not 3.0 specific. i am not sure it has yet been deployed, but can check with the team during our sync.
